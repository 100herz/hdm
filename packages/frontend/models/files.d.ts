export declare interface IFiles {
  id: string
  displayName: string
  path: string
}
