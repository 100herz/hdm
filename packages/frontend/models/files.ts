export interface IFiles {
  id: string
  displayName: string
  path: string
}
