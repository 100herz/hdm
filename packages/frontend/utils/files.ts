/**
 * Open the 'SaveAs' dialog to download a file.
 *
 * @param fileName The file to download.
 */
export function downloadFile(fileName: string) {
  const tempWindow = window.open(fileName)

  if (!tempWindow) return

  tempWindow.document.execCommand('SaveAs')
  tempWindow.close()

  const save = document.createElement('a')
  save.href = fileName
  save.target = '_blank'
  save.download = fileName.substring(fileName.lastIndexOf('/') + 1)

  if (navigator.userAgent.toLowerCase().match(/(ipad|iphone|safari)/) && navigator.userAgent.search('Chrome') < 0) {
    document.location.href = save.href
  } else {
    const evt = new MouseEvent('click', {
      view: window,
      bubbles: true,
      cancelable: false,
    })
    save.dispatchEvent(evt)
    ;(window.URL || window.webkitURL).revokeObjectURL(save.href)
  }
}
