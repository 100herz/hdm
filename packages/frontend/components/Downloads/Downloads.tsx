import { IFiles } from '@/models/files'
import styled from '@emotion/styled'
import dynamic from 'next/dynamic'

interface DownloadsProps extends React.HTMLAttributes<HTMLElement> {
  files: IFiles[]
  title: string
}

export default function Downloads({ files, title, className = '' }: DownloadsProps) {
  const Pdf = dynamic(() => import('@/components/Utils/Pdf'), { ssr: false })

  return (
    <SectionStyled id="kontakt" className={`container lg:px-32 ${className}`}>
      <h3 className="mt-24 text-2xl font-bold lg:text-3xl">{title}</h3>

      <div className="mt-8 download-list">
        {files.map(file => (
          <div key={file.id} className="lg:text-lg download-list__item">
            <Pdf title={file.displayName} filename={file.path} buttonStyleLight buttonWidthFull />
          </div>
        ))}
      </div>
    </SectionStyled>
  )
}

const SectionStyled = styled.section`
  display: block;

  .download-list {
    &__item {
      margin-top: 1rem;
    }

    @media (min-width: 1024px) {
      display: flex;
      flex-wrap: wrap;
      width: 100%;

      &__item {
        flex: 40%;
        margin-top: 2rem;

        &:nth-of-type(even) {
          margin-left: 2rem;
        }
      }
    }
  }
`
