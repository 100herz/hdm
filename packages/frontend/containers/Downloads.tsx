import Downloads from '@/components/Downloads/Downloads'
import Layout from '@/components/Layout'
import certificatesCeDop from '@/data/ce-dop.json'
import certificatesFscPefc from '@/data/fsc-pefc.json'
import { IFiles } from '@/models/files'

export default function DownloadContainer() {
  return (
    <Layout title="Downloads * HDM &ndash; Holz Design Moers">
      <h2 className="container mt-12 text-4xl font-bold lg:mt-32 lg:px-32 lg:text-5xl">Downloads</h2>

      <Downloads title="FSC / PEFC" files={certificatesFscPefc as IFiles[]} />

      <Downloads title="CE-DOP Leistungsbeschreibungen" files={certificatesCeDop as IFiles[]} />
    </Layout>
  )
}
