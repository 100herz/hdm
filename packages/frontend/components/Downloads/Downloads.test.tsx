import { render } from '@/test/test-utils'

import Downloads from './Downloads'
import { IFiles } from '@/models/files'

const dummyFiles: IFiles[] = [
  {
    id: '1',
    displayName: 'DUMMY NAME',
    path: 'DUMMY PATH',
  },
]

describe('Downloads component', () => {
  it('matches snapshot', () => {
    const { container, getByText } = render(<Downloads title="FSC / PEFC" files={dummyFiles} />)
    expect(getByText('FSC / PEFC')).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})
