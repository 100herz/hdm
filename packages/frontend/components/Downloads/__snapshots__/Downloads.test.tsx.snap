// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Downloads component matches snapshot 1`] = `
.emotion-0 {
  display: block;
}

.emotion-0 .download-list__item {
  margin-top: 1rem;
}

@media (min-width: 1024px) {
  .emotion-0 .download-list {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-flex-wrap: wrap;
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
    width: 100%;
  }

  .emotion-0 .download-list__item {
    -webkit-flex: 40%;
    -ms-flex: 40%;
    flex: 40%;
    margin-top: 2rem;
  }

  .emotion-0 .download-list__item:nth-of-type(even) {
    margin-left: 2rem;
  }
}

<section
  class="container lg:px-32  emotion-0 emotion-1"
>
  <h3
    class="mt-24 text-2xl font-bold lg:text-3xl"
  >
    FSC / PEFC
  </h3>
  <div
    class="mt-8 download-list"
  >
    <div
      class="lg:text-lg download-list__item"
    />
  </div>
</section>
`;
